source/frontend/src/features/mapSideBar/compensation/detail/CompensationRequisitionHistoricalDetailContainer.tsx

```diff
@@ -0,0 +1,350 @@
+  } = useProductProvider();
+
+  const fetchParentFile = useCallback(async () => {
+    if (fileType === ApiGen_CodeTypes_FileTypes.Acquisition) {
+      const response = await getAcquisition(parentFileId, adjustedTime);
+      setParentFile(response);
+    } else if (fileType === ApiGen_CodeTypes_FileTypes.Lease) {
+      const response = await getLease(parentFileId, adjustedTime);
+      setParentFile(response);
+    }
+  }, [parentFileId, fileType, adjustedTime, getAcquisition, getLease]);
+
+  const fetchProject = useCallback(
+    async (
+      projectId: number,
+      setter: (value: React.SetStateAction<ApiGen_Concepts_Project>) => void,
+    ) => {
+      const response = await getProject(projectId, adjustedTime);
+      setter(response);
+    },
+    [adjustedTime, getProject],
+  );
+
+  const fetchProduct = useCallback(
+    async (product: number) => {
+      setProduct(null);
+      const response = await getProduct(product, adjustedTime);
+      setProduct(response);
+    },
+    [adjustedTime, getProduct],
+  );
+
+  const fetchCompensationRequisition = useCallback(async () => {
+    setTimedCompensation(null);
+    const response = await getCompensationRequisition(compensationRequisitionId, adjustedTime);
+    setTimedCompensation(response);
+  }, [compensationRequisitionId, adjustedTime, getCompensationRequisition]);
+
+  const fetchCompensationProperties = useCallback(async () => {
+    if (fileType === ApiGen_CodeTypes_FileTypes.Acquisition) {
+      const compReqProperties = await getCompensationAcqProperties(
+        compensationRequisitionId,
+        adjustedTime,
+      );
+      setCompensationRequisitionProperties(compReqProperties ?? []);
+    }
+    if (fileType === ApiGen_CodeTypes_FileTypes.Lease) {
+      const compReqProperties = await getCompensationLeaseProperties(
+        compensationRequisitionId,
+        adjustedTime,
+      );
+      setCompensationRequisitionProperties(compReqProperties ?? []);
+    }
+  }, [
+    compensationRequisitionId,
+    fileType,
+    getCompensationAcqProperties,
+    adjustedTime,
+    getCompensationLeaseProperties,
+  ]);
+
+  const fetchCompensationPayees = useCallback(async () => {
+    if (fileType === ApiGen_CodeTypes_FileTypes.Acquisition) {
+      const compReqAcqPayees = await getCompensationAcqPayees(
+        compensationRequisitionId,
+        adjustedTime,
+      );
+      setAcquisitionPayees(compReqAcqPayees);
+    } else if (fileType === ApiGen_CodeTypes_FileTypes.Lease) {
+      const compReqLeasePayees = await getCompensationLeasePayees(
+        compensationRequisitionId,
+        adjustedTime,
+      );
+      setLeasePayees(compReqLeasePayees);
+    }
+  }, [
+    compensationRequisitionId,
+    fileType,
+    getCompensationAcqPayees,
+    adjustedTime,
+    getCompensationLeasePayees,
+  ]);
+
+  const fetchFinancialCodes = useCallback(async () => {
+    const fetchFinancialActivitiesCall = fetchFinancialActivities();
```

Comment:
These fetches don't seem to be using an *atTime method, but I know on the backend one of the repositories was...  is that an issue?

Reply:
AFAIK the code tables do not change, but instead the code is disabled.

Reply:
Based on my testing they can change (at least functionally, maybe not via business process).